app/app.pro

```diff
@@ -185,17 +185,26 @@ linux {
     INSTALLS += bashcompletion zshcompletion target mimepackage desktopentry icon
 }
 
-
-
 # --- core_lib ---
-win32:CONFIG(release, debug|release): LIBS += -L$$OUT_PWD/../core_lib/release/ -lcore_lib
-else:win32:CONFIG(debug, debug|release): LIBS += -L$$OUT_PWD/../core_lib/debug/ -lcore_lib
-else:unix: LIBS += -L$$OUT_PWD/../core_lib/ -lcore_lib
 
 INCLUDEPATH += $$PWD/../core_lib/src
+DEPENDPATH += $$PWD/../core_lib/src
+
+CONFIG(debug,debug|release) BUILDTYPE = debug
+CONFIG(release,debug|release) BUILDTYPE = release
+
+win32-msvc*{
+  LIBS += -L$$OUT_PWD/../core_lib/$$BUILDTYPE/ -lcore_lib
+  PRE_TARGETDEPS += $$OUT_PWD/../core_lib/$$BUILDTYPE/core_lib.lib
+}
+
+win32-g++{
+  LIBS += -L$$OUT_PWD/../core_lib/ -lcore_lib
+  PRE_TARGETDEPS += $$OUT_PWD/../core_lib/libcore_lib.a
```

Comment:
Don't know why, but MinGW does need the `$$BUILDTYPE` in lib paths on my PC. It turns ou no need to differentiate MSVC and G++. I am using Qt 5.12.7 and MinGW 7.3.0.

Reply:
And that's the thing I found weird.. it seems to be inconsistent, because on my windows setup, it doesn't. so there's still a problem as far as I can see.. maybe we should just force release and debug folders allover and be done with it